opentracing-jms-2/src/test/java/io/opentracing/contrib/jms2/TracingArtemisViaConnectionFactoryTest.java

```diff
@@ -0,0 +1,189 @@
+
+    HashSet<TransportConfiguration> transports = new HashSet<>();
+    transports.add(new TransportConfiguration(InVMAcceptorFactory.class.getName()));
+    configuration.setAcceptorConfigurations(transports);
+    configuration.setSecurityEnabled(false);
+
+    File targetDir = new File(System.getProperty("user.dir") + "/target");
+    configuration.setBrokerInstance(targetDir);
+
+    server = new ActiveMQServerImpl(configuration);
+    server.start();
+
+    ConnectionFactory activeMQJMSConnectionFactory = new ActiveMQJMSConnectionFactory("vm://0");
+    ConnectionFactory tracingConnectionFactory = new TracingConnectionFactory(activeMQJMSConnectionFactory, mockTracer);
+    connection = tracingConnectionFactory.createConnection();
+    connection.start();
+
+    session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+
+    jmsContext = tracingConnectionFactory.createContext();
+  }
+
+  @After
+  public void after() throws Exception {
+    jmsContext.close();
+    session.close();
+    connection.close();
+    server.stop();
+  }
+
+  @Test
+  public void sendAndReceive() throws Exception {
+    Queue queue = session.createQueue("TEST.FOO");
+    MessageProducer producer = session.createProducer(queue);
+    MessageConsumer consumer = session.createConsumer(queue);
+    TextMessage message = session.createTextMessage("Hello world");
+
+    producer.send(message);
+
+    TextMessage received = (TextMessage) consumer.receive(5000);
+    assertEquals("Hello world", received.getText());
+
+    List<MockSpan> mockSpans = mockTracer.finishedSpans();
+    assertEquals(2, mockSpans.size());
+
+    checkSpans(mockSpans);
+    assertNull(mockTracer.activeSpan());
+  }
+
+  @Test
+  public void sendAndReceiveJMSProducer() throws Exception {
+    Destination destination = session.createQueue("TEST.FOO");
+    JMSProducer producer = jmsContext.createProducer();
+    MessageConsumer consumer = session.createConsumer(destination);
+    TextMessage message = session.createTextMessage("Hello world");
+
+    producer.send(destination, message);
+
+    TextMessage received = (TextMessage) consumer.receive(5000);
+    assertEquals("Hello world", received.getText());
+
+    List<MockSpan> mockSpans = mockTracer.finishedSpans();
+    assertEquals(2, mockSpans.size());
+
+    checkSpans(mockSpans);
+    assertNull(mockTracer.activeSpan());
+  }
+
+  @Test
+  public void sendAndReceiveInListener() throws Exception {
+    Destination destination = session.createQueue("TEST.FOO");
+    MessageProducer producer = session.createProducer(destination);
+    MessageConsumer consumer = session.createConsumer(destination);
+
+    final CountDownLatch countDownLatch = new CountDownLatch(1);
+    MessageListener listener = new MessageListener() {
+      @Override
+      public void onMessage(Message message) {
+        countDownLatch.countDown();
+      }
+    };
+    consumer.setMessageListener(listener);
```

Comment:
Btw, this listener isn't instrumented.
For some reason, `setMessageListener` written differently:

https://github.com/opentracing-contrib/java-jms/blob/3aa4af830745624e9bb593e6bd6b6a08e6bddbcc/opentracing-jms-2/src/main/java/io/opentracing/contrib/jms2/TracingSession.java#L128-L131
https://github.com/opentracing-contrib/java-jms/blob/3aa4af830745624e9bb593e6bd6b6a08e6bddbcc/opentracing-jms-common/src/main/java/io/opentracing/contrib/jms/common/TracingMessageConsumer.java#L63-L70